### PR TITLE
Fix/broken anchors docs

### PR DESCRIPTION
**Pull request by banasa44**
## Description

As per the update of Docusaurus in the Developer-Portal, SDK jsdoc2md auto-documentation-generator framework was generating broken anchors error in the Developer-Portal. 

jsdoc2md works with [default templates](https://github.com/jsdoc2md/dmd) that can be override and extended. In this PR are some added partial and helper functions and some overridden. A library had to be added for some of the functions ([array-back](https://www.npmjs.com/package/array-back?activeTab=readme)).

There's also a syntax fix in some MD reference links which wasn't properly redirecting within the documentation.

Task ID: [OS-1074](https://aragonassociation.atlassian.net/browse/OS-1074)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them when possible.
- [ ] I have updated the `CHANGELOG.md` file in the root folder of the package after the `[UPCOMING]` title and before the latest version.
- [ ] I have tested my code on the test network.

[OS-1074]: https://aragonassociation.atlassian.net/browse/OS-1074?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/tweaks/custom-link.hbs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+{{#link to~}}
+{{#if url~}}
+[<code>{{#if ../../caption}}{{escape ../../../caption}}{{else}}{{escape name}}{{/if}}</code>]({{{toLowerCase url}}})
+{{~else~}}
+{{#if ../../caption}}{{escape ../../../caption}}{{else}}{{{escape (toLowerCase (extractText name))}}}{{/if~}}
+{{~/if~}}
+{{/link~}}
+
+{{/if~}}
```
Comment (Rekard0) [inline docs/tweaks/custom-link.hbs]:
```suggestion
{{/if~}}

```

File: docs/tweaks/custom-linked-type-list.hbs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{{#each types~}}
+{{>custom-link to=this html=../html ~}}
+{{#unless @last}}{{{../delimiter}}}{{/unless~}}
+{{/each}}
```
Comment (Rekard0) [inline docs/tweaks/custom-linked-type-list.hbs]:
```suggestion
{{/each}}

```

File: docs/tweaks/ddata.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          .replace(/~/g, '..')
+          .replace(/\(\)/g, '_new')
+          .replace(/#/g, '')
+          .replace(/\./g, '')
+          .toLowerCase(), 
+          methodSign
+      )
+      
+  }
+
```
Comment (Rekard0) [inline docs/tweaks/ddata.js]: @josemarinas why this is happening with JS file? can u look into it

File: docs/tweaks/sig-link.hbs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+{{#if @prefix}}{{@prefix}} {{/if~}}
+{{@accessSymbol}}{{#if (isEvent)}}"{{{name}}}"{{else}}{{{name}}}{{/if~}}
+{{~#if @methodSign}}{{#if (isEvent)}} {{@methodSign}}{{else}}{{@methodSign}}{{/if}}{{/if~}}
+{{{@codeClose}}}](#{{{anchorName}}}{{#if @returnTypes}}--{{>custom-linked-type-list types=@returnTypes delimiter="--" }}{{/if~}})
+{{~#if @returnSymbol}} {{@returnSymbol}}{{/if~}}
+{{#if @returnTypes}} {{>linked-type-list types=@returnTypes delimiter=" \| " }}{{/if~}}
+{{#if @suffix}} {{@suffix}}{{/if~}}
+{{{@depClose}~}}
+{{~/sig}}{{/if~}}
+{{#if virtual}}*{{/if}}{{#with (parentObject)}}{{#if virtual}}*{{/if~}}{{/with~}}
```
Comment (Rekard0) [inline docs/tweaks/sig-link.hbs]:
```suggestion
{{#if virtual}}*{{/if}}{{#with (parentObject)}}{{#if virtual}}*{{/if~}}{{/with~}}

```

File: docs/tweaks/ddata.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+var util = require('util')
```
Comment (josemarinas) [inline docs/tweaks/ddata.js]: please fix the name of this file

File: docs/tweaks/ddata.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+var arrayify = require('array-back')
+exports.anchorName = anchorName
+
+function anchorName(options) {
+    var methodSign = options.data.methodSign ? options.data.methodSign.replace(/[^\w,]/g, '-').replace(/^-+|-+$/g, '').replace(/-+/g, '').replace(/,/g, '-').toLowerCase() : '';
```
Comment (josemarinas) [inline docs/tweaks/ddata.js]: Noted in the sonar

imagen

Comment (josemarinas) [inline docs/tweaks/ddata.js]: use let for variables const for constants

File: docs/tweaks/ddata.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+    if (!this.id) throw new Error('[anchorName helper] cannot create a link without a id: ' + JSON.stringify(this))
+    if (this.inherited) {
+      options.hash.id = this.inherits
+      var inherits = _identifier(options)
```
Comment (josemarinas) [inline docs/tweaks/ddata.js]: same, `let` for variables `const` for constats, `var` days are over

File: docs/tweaks/ddata.js (hunk 5 above)
Comment (josemarinas) [inline docs/tweaks/ddata.js]: same

File: docs/tweaks/ddata.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+var util = require('util')
+var arrayify = require('array-back')
```
Comment (josemarinas) [inline docs/tweaks/ddata.js]: same

File: docs/tweaks/ddata.js (hunk 5 above)
Comment (banasa44) [reply docs/tweaks/ddata.js]: This is overriding functions of a template default file therefore the name has to be the same.
https://github.com/jsdoc2md/dmd/blob/master/helpers/ddata.js

File: docs/tweaks/ddata.js (hunk 3 above)
Comment (josemarinas) [reply docs/tweaks/ddata.js]: there is an empty or tabulation after the line jump. If you delete it it should be good

File: package.json [hunk 9]
```diff
@@ -28,10 +28,13 @@
...
   "devDependencies": {
-    "@babel/preset-typescript": "^7.18.6",
     "@babel/plugin-proposal-object-rest-spread": "^7.20.7",
+    "@babel/preset-typescript": "^7.18.6",
     "jsdoc-babel": "^0.5.0",
     "jsdoc-to-markdown": "^8.0.1",
     "turbo": "^1.1.9"
+  },
+  "dependencies": {
+    "array-back": "^6.2.2"
```
Comment (josemarinas) [inline package.json]: is only used for docs, add it to dev depenndencies instead

Comment (banasa44) [reply package.json]: I deleted it as it was unused.

**Review by Rekard0 — APPROVED** [review]

**Pull request merged**